### PR TITLE
Bump pimple/pimple to 3.5.0

### PR DESCRIPTION
**Pull request by come-nc**
Signed-off-by: Côme Chilliet <come.chilliet@nextcloud.com>

### CODE REVIEW

**Comment by CarlSchwan** [comment]
Looks good from my side :) Not sure why we didn't need git clean before but it makes sense.

**Comment by come-nc** [comment]
@ChristophWurst @nickvergessen Can you validate that my CI fix is okay?
The CI has been failing for a week because the tests get inserted in the autoload.

**Comment by nickvergessen** [comment]
> Can you validate that my CI fix is okay?

But that e.g. would have removed the symfony subdependency in https://github.com/nextcloud/3rdparty/pull/903/files you forgot to add?

**Comment by come-nc** [comment]
> 



> > Can you validate that my CI fix is okay?
> 
> But that e.g. would have removed the symfony subdependency in https://github.com/nextcloud/3rdparty/pull/903/files you forgot to add?

No, because git clean -X only deletes files ignored by the .gitignore, which is not the case of new dependencies.

File: .github/workflows/composer.yml [hunk 1]
```diff
@@ -26,6 +26,8 @@ jobs:
       run: rm -rf ./*/
     - name: Download dependencies
       run: composer install --no-dev
+    - name: Remove tests and other untracked files
+      run: git clean -X -d -f
```
Comment (ChristophWurst) [inline .github/workflows/composer.yml:30]: this would revert https://github.com/nextcloud/3rdparty/pull/815

there was an issue with xdf

**Review by ChristophWurst — CHANGES_REQUESTED** [review]

File: .github/workflows/composer.yml (hunk 1 above)
Comment (come-nc) [reply .github/workflows/composer.yml:30]: Wait, so the tests are supposed to be in the autoload then?

Comment (ChristophWurst) [reply .github/workflows/composer.yml:30]: IMO no, they shouldn't. But I lack the context on the other PR. Just linked it to show xdf was used before and it was intentionally removed :)

**Comment by come-nc** [comment]
Commit https://github.com/nextcloud/3rdparty/commit/2ae1a1d6f688ae8394d6559ee673fecbee975db4 has removed tests from the autoload and all commits since have been failing CI.

File: .github/workflows/composer.yml (hunk 1 above)
Comment (come-nc) [reply .github/workflows/composer.yml:30]: Actually before #815 it was `git clean -x` which would remove any new file. My commit uses `git clean -X` which only removes files covered by .gitignore
So it should be fine IMO.

**Review by ChristophWurst — APPROVED** [review]

**Pull request merged**